io-engine/src/grpc/v1/snapshot.rs

```diff
@@ -599,133 +599,17 @@ impl SnapshotRpc for SnapshotService {
-            async move {
-                let args = request.into_inner();
-                info!("{:?}", args);
-                let rx = rpc_submit(async move {
-                    let snap_lvol = match UntypedBdev::lookup_by_uuid_str(
-                        &args.snapshot_uuid,
-                    ) {
-                        Some(bdev) => Lvol::try_from(bdev)?,
-                        None => {
-                            return Err(LvsError::Invalid {
-                                source: Errno::ENOENT,
-                                msg: format!(
-                                    "Snapshot {} not found",
-                                    args.snapshot_uuid
-                                ),
-                            })
-                        }
-                    };
-                    if UntypedBdev::lookup_by_uuid_str(&args.clone_uuid).is_some() {
-                        return Err(LvsError::Invalid {
-                            source: Errno::EEXIST,
-                            msg: format!(
-                                "clone uuid {} already exist",
-                                args.clone_uuid
-                            ),
-                        })
-                    };
-                    // prepare clone config.
-                    let clone_config =
-                        match snap_lvol.prepare_clone_config(
-                            &args.clone_name,
-                            &args.clone_uuid,
-                            &args.snapshot_uuid
-                        ) {
-                            Some(clone_config) => clone_config,
-                            None => return Err(LvsError::Invalid {
-                                source: Errno::EINVAL,
-                                msg: format!(
-                                    "Invalid parameters clone_uuid: {}, clone_name: {}",
-                                    args.clone_uuid,
-                                    args.clone_name
-                                ),
-                            })
-                        };
-                    // create clone
-                    match snap_lvol.create_clone(clone_config).await {
-                        Ok(clone_lvol) => {
-                            info!("Create Clone Success for {snap_lvol:?}, {clone_lvol:?}");
-                            Ok(Replica::from(clone_lvol))
-                        }
-                        Err(e) => {
-                            error!(
-                                "Create clone Failed for snapshot: {snap_lvol:?} with Error: {e:?}",
-                            );
-                            Err(e)
-                        }
-                    }
-                })?;
-                rx.await
-                    .map_err(|_| Status::cancelled("cancelled"))?
-                    .map_err(Status::from)
-                    .map(Response::new)
-            },
-        )
-        .await
+        todo!("Not implemented");
```

Comment:
not this todo, this will panic!
Have to result GrpcResult with status unimplemented

Reply:
Might be better to remove the service endpoints the mayastor-api protobuf completely, then we just remove these completely.

Reply:
Right. Removed clone related gRPC request.